### PR TITLE
Create a unit test for Update Device Service LastReported by ID.

### LINKED ISSUE #1737: Unit Test -- Core-Metadata: Create a unit test for Update Device Service LastReported by ID.
Create a unit test at the REST layer level (add to `internal/core/metadata/rest_deviceservice_test.go`).

### PR DESCRIPTION
**Pull request by akramtexas**
Fix #1737

### CODE REVIEW

File: internal/core/metadata/rest_deviceservice_test.go [hunk 1]
```diff
@@ -594,3 +691,20 @@ func createDeviceServiceRequestWithBody(
...
+	return dbMock
+}
+
+func newMockDBClientSecond() interfaces.DBClient {
+	dbMock := &mocks.DBClient{}
+	dbMock.On("GetDeviceServiceById", testDeviceServiceLastUpdatedId).Return(testDeviceServiceLastUpdated, nil)
+	dbMock.On("UpdateDeviceService", mock.Anything).Return(testError)
+
+	return dbMock
+}
```
Comment (AnthonyMBonafide) [inline internal/core/metadata/rest_deviceservice_test.go]: Can this be removed in favor of refactoring `createMockWithOutlines` to support mocking methods with varying number or return entities? It seems like we have outgrown the original function and it needs to be updated to support this. If we do want to keep these functions I think we should rename them better so it is clear what the difference is between them.

If we do refactor the `createMockWithOutlines` I think we can than 
 remove `TestUpdateServiceLastReportedByIdServiceUnavailable` in favor of a test table.

File: internal/core/metadata/rest_deviceservice_test.go [hunk 2]
```diff
@@ -560,6 +576,87 @@ func TestUpdateAdminStateByName(t *testing.T) {
...
+			createMockWithOutlines([]mockOutline{
+				{"GetDeviceServiceById", testDeviceServiceLastUpdatedNotExistId, contract.DeviceService{}, db.ErrNotFound},
+			}),
+			http.StatusInternalServerError,
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			rr := httptest.NewRecorder()
+			var loggerMock = logger.NewMockClient()
```
Comment (AnthonyMBonafide) [inline internal/core/metadata/rest_deviceservice_test.go]: For consistency, can we use the `:=` operator here. Our code base tends to favor `:=` over `var`. This also applies to other uses of `var` in these changes.

File: internal/core/metadata/rest_deviceservice_test.go (hunk 1 above)
Comment (akramtexas) [reply internal/core/metadata/rest_deviceservice_test.go]: I like the suggestion and will pursue the implementation 👍

**Comment by akramtexas** [comment]
@AnthonyMBonafide 
- Following up on your suggestion, I have refactored the func `createMockWithOutlines()` to support mocking methods with varying number of arguments and/or varying number of return entities. This resulted in updating about 50+ locations wherefrom `createMockWithOutlines()` is called. 
- Removed TestUpdateServiceLastReportedByIdServiceUnavailable() in favor of a table-driven test.

**Pull request merged**